samples/coop_synchronization/src/main.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2012-2014 Wind River Systems, Inc.
+ * Copyright (c) 2023 Advanced Micro Devices, Inc. (AMD)
+ * Copyright (c) 2023 Alp Sayin <alpsayin@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/printk.h>
+
+/*
+ * The coop synchronization demo has two threads that utilize semaphores and k_yield
+ * to take turns printing a greeting message with a somewhat uncontrolled rate.
+ * The rate is controlled via a busy waiting loop. The advantage of this demo is that
+ * it'll work even if there are no timing utilities provided by the architecture.
+ */
+
+/* size of stack area used by each thread */
+#define STACKSIZE 1024
+
+/* number of NOPs to churn through for busy waiting */
+#if defined(CONFIG_TIMING_FUNCTIONS)
+#define NUM_NOPS() timing_freq_get()
+#else
+#define NUM_NOPS() (1000000)
+#endif
+
+/* arch_curr_cpu() is defined only for certain architectures */
+#if defined(CONFIG_ARC) || defined(CONFIG_ARM) || defined(CONFIG_X86_64) \
+	|| defined(CONFIG_RISCV) || defined(CONFIG_XTENSA)
+	#define GET_CURR_CPU() arch_curr_cpu()->id
+#else
+	#define GET_CURR_CPU() (0)
+#endif
+
+/*
+ * @param my_name      thread identification string
+ * @param my_sem       thread's own semaphore
+ * @param other_sem    other thread's semaphore
+ */
+void helloLoop(const char *my_name, struct k_sem *my_sem,
+	struct k_sem *other_sem, int incrementer)
+{
+	const char *tname;
+	uint8_t cpu = 0;
+	struct k_thread *current_thread;
+	int task_local_counter = 0;
+
+	while (1) {
+		/* take my semaphore */
+		k_sem_take(my_sem, K_FOREVER);
+
+		current_thread = k_current_get();
+		tname = k_thread_name_get(current_thread);
+		cpu = GET_CURR_CPU();
+
+		/* say "hello" */
+		if (tname == NULL) {
+			printk("%s: Hello World from cpu %d on %s! counter = %d\n", my_name, cpu,
+						 CONFIG_BOARD, task_local_counter += incrementer);
+		} else {
+			printk("%s: Hello World from cpu %d on %s! counter = %d\n", tname, cpu,
+						 CONFIG_BOARD, task_local_counter += incrementer);
+		}
```

Comment:
Probably we can simplify it to:

```suggestion
		printk("%s: Hello World from cpu %d on %s! counter = %d\n", tname == NULL ? my_name : tname,
						 cpu, CONFIG_BOARD, task_local_counter += incrementer);
```

Or even this way:
```suggestion
		printk("%s: Hello World from cpu %d on %s! counter = %d\n", tname ?: my_name, cpu,
						 CONFIG_BOARD, task_local_counter += incrementer);
```

Reply:
I prefer the top because of the explicitness.